test/helper.rb

```diff
@@ -80,17 +80,48 @@ def self.call(host = '127.0.0.1')
 end
 
 require "timeout"
-module TimeoutEveryTestCase
-  # our own subclass so we never confuse different timeouts
-  class TestTookTooLong < Timeout::Error
+
+if Minitest::VERSION < '5.25'
+  module TimeoutEveryTestCase
+    # our own subclass so we never confuse different timeouts
+    class TestTookTooLong < Timeout::Error
+    end
+
+    def run
+      with_info_handler do
+        time_it do
+          capture_exceptions do
+            ::Timeout.timeout($test_case_timeout, TestTookTooLong) do
+              before_setup; setup; after_setup
+              self.send self.name
+            end
+          end
+
+          capture_exceptions do
+            ::Timeout.timeout($test_case_timeout, TestTookTooLong) do
+              Minitest::Test::TEARDOWN_METHODS.each { |hook| self.send hook }
+            end
+          end
+          if respond_to? :clean_tmp_paths
+            clean_tmp_paths
+          end
+        end
+      end
+
+      Minitest::Result.from self # per contract
+    end
   end
+else
+  module TimeoutEveryTestCase
+    # our own subclass so we never confuse different timeouts
+    class TestTookTooLong < Timeout::Error
+    end
 
-  def run
-    with_info_handler do
+    def run
```

Comment:
This doesn't make sense to me. You shouldn't be calling with_info_handler... and you prolly shouldn't be overriding run like this, let alone having to duplicate all of its code. Instead, override [Minitest::Runnable.run_one_method](https://github.com/minitest/minitest/blob/master/lib/minitest.rb#L444) and you don't need ANY of this boilerplate. Should look something like this:

```ruby
def run_one_method klass, method_name, reporter
  ::Timeout.timeout($test_case_timeout, TestTookTooLong) do
    super
  end
ensure
  clean_tmp_paths if respond_to? :clean_tmp_paths
end
```

because it is a class method, you'll prolly have to `extend` rather than `include`... I would make it a subclass of Runnable and subclass it. Happy to provide a PR if you want to see it in full.

Reply:
The code for this dates back many years.  Thanks for the suggestion.  I haven't checked, we may have needed `clean_tmp_paths` to run after the 'teardown' methods.  Not.sure.  Some files we may check whether Puma removed them, so `clean_tmp_paths` is just for cleanup after **all the test code* runs...

Reply:
After refreshing, we could probably use `clean_tmp_paths` in `after_teardown`...